src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1115,50 +1073,85 @@ void CASESession::SendErrorMsg(SigmaErrorType errorCode)
                    ChipLogError(SecureChannel, "Failed to send error message"));
 }
 
-CHIP_ERROR CASESession::FindValidTrustedRoot(const System::PacketBufferTLVReader & tlvReader, uint32_t nTrustedRoots)
+CHIP_ERROR CASESession::GenerateDestinationID(const ByteSpan & random, const Credentials::CertificateKeyId * trustedRootId,
+                                              NodeId nodeId, FabricId fabricId, MutableByteSpan & destinationId)
 {
-    CertificateKeyId trustedRoot;
-    System::PacketBufferTLVReader suppTlvReader;
-    uint8_t trustedRootId[kTrustedRootIdSize];
+    HMAC_sha hmac;
+    uint16_t destinationMessageLen = kSigmaParamRandomNumberSize + kP256_PublicKey_Length + sizeof(FabricId) + sizeof(NodeId);
+    System::PacketBufferHandle destinationMessage = System::PacketBufferHandle::New(destinationMessageLen);
```

Comment:
This should not use packet buffers as memory. Use a `ScopedBuffer` instead if heap is needed.

Overall, the size here could be made a constant expression, and the buffer could be done cheaper by using a stack buffer. The length will be 32 + 65 + 8 + 8 = 113 bytes, which is not crazy. It's possibly less stack then the entire alloc chain below.


```
constexpr size_t destinationMessageLen = kSigmaParamRandomNumberSize + kP256_PublicKey_Length + sizeof(FabricId) + sizeof(NodeId);

uint8_t bbuf[destinationMessageLen];
```

Reply:
Updated.